backend/plugin/db/hive/sync.go

```diff
@@ -0,0 +1,130 @@
+package hive
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/pkg/errors"
+
+	"github.com/bytebase/bytebase/backend/plugin/db"
+	storepb "github.com/bytebase/bytebase/proto/generated-go/store"
+)
+
+// TODO(tommy): another approch for this is fetching data from metastore database.
+func (d *Driver) SyncInstance(ctx context.Context) (*db.InstanceMetadata, error) {
+	var instanceMetadata db.InstanceMetadata
+
+	version, err := d.getVersion(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	databaseSchemaMetadata, err := d.SyncDBSchema(ctx)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to sync database schema")
+	}
+
+	// TODO(tommy): roles
+	instanceMetadata.Databases = append(instanceMetadata.Databases, databaseSchemaMetadata)
+	instanceMetadata.Version = version
+	return &instanceMetadata, nil
+}
+
+// It should be noted that Schema and database have the same meaning in Hive.
+func (d *Driver) SyncDBSchema(ctx context.Context) (*storepb.DatabaseSchemaMetadata, error) {
+	databaseSchemaMetadata := new(storepb.DatabaseSchemaMetadata)
+
+	databaseNames, err := d.getDatabaseNames(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	var schemaMetadata []*storepb.SchemaMetadata
+	execOpts := db.ExecuteOptions{}
+	for _, database := range databaseNames {
+		// change database.
+		_, err := d.Execute(ctx, fmt.Sprintf("use %s", database), execOpts)
+		if err != nil {
+			return nil, errors.Wrapf(err, "failed to switch to database %s", database)
+		}
+		// fetch table metadata.
+		tableMetadata, err := d.getTables(ctx)
+		if err != nil {
+			return nil, errors.Wrapf(err, "cannot get table metadata from database %s", database)
+		}
```

Comment:
Be aware of the underlying connection pool.

Reply:
sorry but can you make it more explict? All I know is that there is no such connection pool underlying this gohive driver. 😆